packages/react-reconciler/src/ReactFiberNewContext.new.js

```diff
@@ -130,18 +143,24 @@ export function popProvider(
   providerFiber: Fiber,
 ): void {
   const currentValue = valueCursor.current;
-  pop(valueCursor, providerFiber);
```

Comment:
Moved down since we push(value) -> push(rendererN) so we need to pop(rendererN) -> pop(value)